<a href="https://colab.research.google.com/github/kemalyildirim/helloworld-machinelearning/blob/master/cse315_161805025_assignment_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Kemal Yildirim - 161805025 Adnan Menders University

CSE 315 Assignment#1

UCI's poker hand data set.

For detailed information: https://archive.ics.uci.edu/ml/datasets/Poker+Hand

**Data Set Information:**

Each record is an example of a hand consisting of five playing cards drawn from a standard deck of 52.

**Attirubutes:**

Features consist of five cards; each card have rank(labeled as numerically  1 to 13 (Ace, 2, 3, ... Queen King)) and suit(Hearts, Spades, Diamonds, Clubs)

Output is class of the poker hand labeled as 0 to 9 numerically:


0: Nothing in hand; not a recognized poker hand 

1: One pair; one pair of equal ranks within five cards 

2: Two pairs; two pairs of equal ranks within five cards 

3: Three of a kind; three equal ranks within five cards 

4: Straight; five cards, sequentially ranked with no gaps 

5: Flush; five cards with the same suit 

6: Full house; pair + different rank three of a kind 

7: Four of a kind; four equal ranks within five cards 

8: Straight flush; straight + flush 

9: Royal flush; {Ace, King, Queen, Jack, Ten} + flush 


Import libraries.

In [0]:
import pandas as pd
import numpy as np
import urllib

Fetch data set and name columns.

In [0]:
column = ["S1","C1","S2","C2","S3","C3","S4","C4","S5","C5","CLASS"]
link = "https://archive.ics.uci.edu/ml/machine-learning-databases/poker/"

#Fetch training and test set
urllib.request.urlretrieve(link + "poker-hand-training-true.data","training.csv")
urllib.request.urlretrieve(link + "poker-hand-testing.data","testing.csv")

training_set = pd.read_csv("training.csv", names = column)

#There is a given test set so I will not split the training set into test set.
test_set = pd.read_csv("testing.csv",names = column, engine="python")

In [50]:
print(training_set.head())
print(test_set.head())

   S1  C1  S2  C2  S3  C3  S4  C4  S5  C5  CLASS
0   1  10   1  11   1  13   1  12   1   1      9
1   2  11   2  13   2  10   2  12   2   1      9
2   3  12   3  11   3  13   3  10   3   1      9
3   4  10   4  11   4   1   4  13   4  12      9
4   4   1   4  13   4  12   4  11   4  10      9
   S1  C1  S2  C2  S3  C3  S4  C4  S5  C5  CLASS
0   1   1   1  13   2   4   2   3   1  12      0
1   3  12   3   2   3  11   4   5   2   5      1
2   1   9   4   6   1   4   3   2   3   9      1
3   1   4   3  13   2  13   2   1   3   6      1
4   3  10   2   7   1   2   2  11   4   9      0


(1000000, 11)

Set rows [0:9] as features and row 11 as output:

In [0]:
X_train = training_set.iloc[:,2:-1]
Y_train = training_set['CLASS']

In [71]:
print(X_train.shape)
print(Y_train.shape)

(25010, 8)
(25010,)


There are 25010 poker hands and each hand have 8 features.

Picking first 7500 rows of the test set for testing.(30% of the training set.)

In [69]:
X_test, Y_test = test_set.iloc[:7500,2:-1], test_set.loc[:7500,'CLASS']
print("X test shape: %s", X_test.shape)
print("Y test shape: %s" ,Y_test.shape)
print(X_test.head())
print(Y_test.head())

X test shape: %s (7500, 8)
Y test shape: %s (7501,)
   S2  C2  S3  C3  S4  C4  S5  C5
0   1  13   2   4   2   3   1  12
1   3   2   3  11   4   5   2   5
2   4   6   1   4   3   2   3   9
3   3  13   2  13   2   1   3   6
4   2   7   1   2   2  11   4   9
0    0
1    1
2    1
3    1
4    0
Name: CLASS, dtype: int64


Empty DataFrame
Columns: []
Index: [0, 1, 2, 3, 4]